backend/src/tools/tool_factory.py

```
@@ -0,0 +1,302 @@
+
+from .http_client import get_wrapper, make_api_call_func, fetch_openapi_spec
+from .schema_resolver import resolve_ref, resolve_ref_recursive, create_schema
+from src.utils.logger import logger
+
+
+def construct_api_tool(
+    name: str,
+    description: str,
+    method: str,
+    headers: Dict[str, Any],
+    metadata: Dict[str, Any] = {},
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)